src/irr/asset/CCPUMeshPacker.h

```diff
@@ -86,18 +89,35 @@ PackedMeshBufferData CCPUMeshPacker<MDIStructType>::alloc(const ICPUMeshBuffer c
 
 	//TODO: divide into multiple mdi structs if(idxCnt > m_maxIndexCountPerMDIData)
 	MDIStructType* mdiBuffPtr = static_cast<MDIStructType*>(m_outMDIData.get()->getPointer());
-	*(mdiBuffPtr + MDIAllocAddr) =
+
+	for (int i = 0; i < MDIStructsNeeded; i++)
 	{
-		static_cast<uint32_t>(meshBuffer->getIndexCount()),
-		static_cast<uint32_t>(meshBuffer->getInstanceCount()),
-		idxAllocAddr * sizeof(uint16_t),
-		0,
-		static_cast<uint32_t>(meshBuffer->getBaseInstance())
-	};
+		uint32_t idxCnt = 0u;
+
+		if (i == MDIStructsNeeded - 1)
+		{
+			idxCnt = static_cast<uint32_t>(meshBuffer->getIndexCount()) % maxIdxCntPerPatch;
+
+			if (idxCnt == 0)
+				idxCnt = maxIdxCntPerPatch;
+		}
+		else
+		{
+			idxCnt = maxIdxCntPerPatch;
+		}
 
+		*(mdiBuffPtr + MDIAllocAddr + i) =
+		{
+			idxCnt,
+			static_cast<uint32_t>(meshBuffer->getInstanceCount()),
+			idxAllocAddr + i * maxIdxCntPerPatch,
+			0 /*TODO*/,
+			static_cast<uint32_t>(meshBuffer->getBaseInstance()) /*TODO*/
+		};
+	}
```

Comment:
this you want to do during commit, since any splitting, clustering, etc. will take place then.

Also moving this to `commit` will allow you defer the creation of MDI CPUBuffer (or any buffer at all) until you got your `allocs` all figured out